qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -44,77 +42,98 @@
+            if job_status is JobStatus.ERROR:
+                print("There was en error in th
+
+
+        except JobError as ex:
+            log("Something wrong happened!: {}".format(ex))
+
+    About the status in case of Error or Exception
+    ==============================================
+    If there's an exception from the IBMQuantumExperience API once a job has been submitted,
+    we cannot set the status of the job to ERROR, because it might happen that the job is actually
+    queued or running, and let's say that the exception is caused because of a temporary
+    out of service situation of the server, once this situation is restored, we might want to ask
+    again about the status of our job, that could be, for example, in state: queued, running or
+    done. If something wrong happened to the job, it's server responsability to answer an error
+    response to a status query from Qiskit. But, if the exception is thrown before having a
+    valid job ID, there's nothing we can do with this job instance, as we cannot query the server
+    for information using this job instance.
+
     Attributes:
         _executor (futures.Executor): executor to handle asynchronous jobs
-        _final_states (list(JobStatus)): terminal states of async jobs
     """
     _executor = futures.ThreadPoolExecutor()
 
-    def __init__(self, qobj, api, is_device):
+    def __init__(self, api, is_device, qobj=None, job_id=None, backend_name=None,
+                 creation_date=None):
         """IBMQJob init function.
+        We can instantiate jobs from two sources: A QObj, and an already submitted job returned by
+        the API servers.
 
         Args:
-            qobj (Qobj): job description
             api (IBMQuantumExperience): IBM Q API
             is_device (bool): whether backend is a real device  # TODO: remove this after Qobj
+            qobj (Qobj): The Quantum Object. See notes below
+            job_id (String): The job ID of an already submitted job.
+            backend_name(String): The name of the backend that run the job.
+            creation_date(String): When the job was run.
+
+        Notes:
+            It is mandatory to pass either ``qobj`` or ``job_id``. Passing a ``qobj``
+            will ignore ``job_id`` and will create an instance representing
+            an already-created job retrieved from the API server.
         """
         super().__init__()
-        self._qobj = qobj
+        self._job_data = None
+        # TODO No need for this conversion, just use the new equivalent members above
+        if qobj is not None:
+            old_qobj = qobj_to_dict(qobj, version='0.0.1')
+            self._job_data = {
+                'circuits': old_qobj['circuits'],
+                'hpc':  None if 'hpc' not in old_qobj['config'] else old_qobj['config']['hpc'],
+                'seed': old_qobj['circuits'][0]['config']['seed'],  # TODO <-- [0] ???
```

Comment:
Test would have failed, right? ... Confirming it could probably take time as there are other people involved. Let's write it down and solve it in a follow up.